orchestra/bots/staffbot.py

```diff
@@ -89,17 +122,35 @@ def _send_task_to_worker(self, worker, task):
                     kwargs=url_kwargs))
 
         context = Context({
-            'username': worker.user.username,
+            'username': username,
             'accept_url': accept_url,
             'reject_url': reject_url
         })
 
-        message_body = render_to_string('communication/new_task_available.txt',
-                                        context)
+        message_body = render_to_string(template, context)
+        return message_body
+
+    def send_staffing_request_by_mail(self, staffing_request, message):
+        email = (
+            staffing_request.communication_preference.worker.user.email)
+
         send_mail('New task is available for claim',
-                  message_body,
+                  message,
                   settings.ORCHESTRA_NOTIFICATIONS_FROM_EMAIL,
-                  [worker.user.email])
+                  [email])
+        staffing_request.status = StaffingRequest.Status.SENT.value
+        staffing_request.save()
+
+
+    def send_staffing_request_by_slack(self, staffing_request, message):
+        worker = (
+            staffing_request.communication_preference.worker)
+        if worker.slack_id is None:
+            logger.error('Worker {} does not have a slack id'.format(
+                worker))
+            return
 
+        slack = Slacker(settings.SLACK_EXPERTS_API_KEY)
```

Comment:
slacker is the name of an app. we have been using slack throughout the codebase :)